src/main/java/com/stripe/model/UsageRecord.java

```diff
@@ -16,39 +16,64 @@
 @Setter
 @EqualsAndHashCode(callSuper = false)
 public class UsageRecord extends ApiResource implements HasId {
-  @Getter(onMethod = @__({@Override})) String id;
-  String object;
+  /** Unique identifier for the object. */
+  @Getter(onMethod = @__({@Override}))
+  @SerializedName("id")
+  String id;
+
+  /**
+   * Has the value `true` if the object exists in live mode or the value `false` if the object
+   * exists in test mode.
+   */
+  @SerializedName("livemode")
   Boolean livemode;
+
+  /** String representing the object's type. Objects of the same type share the same value. */
+  @SerializedName("object")
+  String object;
+
+  /** The usage quantity for the specified date. */
+  @SerializedName("quantity")
   Long quantity;
+
+  /** The ID of the subscription item this usage record contains data for. */
+  @SerializedName("subscription_item")
   String subscriptionItem;
+
+  /** The timestamp when this usage occurred. */
+  @SerializedName("timestamp")
   Long timestamp;
 
-  // <editor-fold desc="create">
   /**
-   * Create a new usage record associated with a subscription item.
-   * @param params The quantity, the timestamp and the conflict behaviour (action)
-   * @param options Request options
-   * @return The created usage record
+   * Creates a usage record for a specified subscription item and date, and fills it with a
+   * quantity.
+   *
+   * <p>Usage records provide <code>quantity</code> information that Stripe uses to track how much a
+   * customer is using your service. With usage information and the pricing model set up by the <a
+   * href="https://stripe.com/docs/billing/subscriptions/metered-billing">metered billing</a> plan,
+   * Stripe helps you send accurate invoices to your customers.
+   *
+   * <p>The default calculation for usage is to add up all the <code>quantity</code> values of the
+   * usage records within a billing period. You can change this default behavior with the billing
+   * plan’s <code>aggregate_usage</code> <a
+   * href="https://stripe.com/docs/api/plans/create#create_plan-aggregate_usage">parameter</a>. When
+   * there is more than one usage record with the same timestamp, Stripe adds the <code>quantity
+   * </code> values together. In most cases, this is the desired resolution, however, you can change
+   * this behavior with the <code>action</code> parameter.
+   *
+   * <p>The default pricing model for metered billing is <a
+   * href="https://stripe.com/docs/api/plans/object#plan_object-billing_scheme">per-unit
+   * pricing</a>. For finer granularity, you can configure metered billing to have a <a
+   * href="https://stripe.com/docs/billing/subscriptions/tiers">tiered pricing</a> model.
    */
-  public static UsageRecord create(Map<String, Object> params, RequestOptions options)
+  public static UsageRecord createOnSubscriptionItem(
```

Comment:
3) Create (custom): Static method to create sub-resource on parent resource, taking of parent resource explicitly. 
Contrast this to creating on sub-resource collection relying on the implicit parent resource.

Reply:
https://github.com/stripe/stripe-java/blob/2aa5db5c39dc6e91228507798083f373904cc8c0/src/main/java/com/stripe/model/UsageRecord.java#L69-L79